src/main/java/blusunrize/immersiveengineering/common/data/Recipes.java

```diff
@@ -146,6 +251,143 @@ protected void registerRecipes(@Nonnull Consumer<IFinishedRecipe> out)
+			.build(out);
+
+		ShapedRecipeBuilder.shapedRecipe(WoodenDevices.sorter)
+			.patternLine("wrw")
+			.patternLine("ibi")
+			.patternLine("wcw")
+			.key('i', IETags.getTagsFor(EnumMetals.IRON).ingot)
+			.key('r', Tags.Items.DUSTS_REDSTONE)
+			.key('w', IETags.getItemTag(IETags.treatedWood))
+			.key('b', ConveyorHandler.getBlock(BasicConveyor.NAME))
+			.key('c', Ingredients.componentIron)
+			.addCriterion("has_"+toPath(ConveyorHandler.getBlock(BasicConveyor.NAME)), hasItem(ConveyorHandler.getBlock(BasicConveyor.NAME)))
+			.build(out);
+		ShapedRecipeBuilder.shapedRecipe(WoodenDevices.fluidSorter)
+			.patternLine("wrw")
+			.patternLine("ibi")
+			.patternLine("wcw")
+			.key('i', IETags.getTagsFor(EnumMetals.IRON).ingot)
+			.key('r', Tags.Items.DUSTS_REDSTONE)
+			.key('w', IETags.getItemTag(IETags.treatedWood))
+			.key('b', MetalDevices.fluidPipe)
+			.key('c', Ingredients.componentIron)
+			.addCriterion("has_"+toPath(MetalDevices.fluidPipe), hasItem(MetalDevices.fluidPipe))
+			.build(out);
+
+		ShapedRecipeBuilder.shapedRecipe(WoodenDevices.turntable)
+			.patternLine("iwi")
+			.patternLine("rcr")
+			.key('i', IETags.getTagsFor(EnumMetals.IRON).ingot)
+			.key('c', MetalDecoration.lvCoil)
+			.key('r', Tags.Items.DUSTS_REDSTONE)
+			.key('w', IETags.getItemTag(IETags.treatedWood))
+			.addCriterion("has_"+toPath(MetalDecoration.lvCoil), hasItem(MetalDecoration.lvCoil))
+			.build(out);
+
+		ShapedRecipeBuilder.shapedRecipe(WoodenDevices.windmill)
+			.patternLine("ppp")
+			.patternLine("pip")
+			.patternLine("ppp")
+			.key('p', Ingredients.windmillBlade)
+			.key('i', IETags.getTagsFor(EnumMetals.IRON).ingot)
+			.addCriterion("has_"+toPath(Ingredients.windmillBlade), hasItem(Ingredients.windmillBlade))
+			.build(out);
+		ShapedRecipeBuilder.shapedRecipe(WoodenDevices.watermill)
+			.patternLine(" p ")
+			.patternLine("pip")
+			.patternLine(" p ")
+			.key('p', Ingredients.waterwheelSegment)
+			.key('i', IETags.getTagsFor(EnumMetals.STEEL).ingot)
+			.addCriterion("has_"+toPath(Ingredients.waterwheelSegment), hasItem(Ingredients.waterwheelSegment))
+			.build(out);
+
+		ShapedRecipeBuilder.shapedRecipe(WoodenDevices.gunpowderBarrel)
+			.patternLine(" f ")
+			.patternLine("gbg")
+			.patternLine("ggg")
+			.key('f', Ingredients.hempFiber)
+			.key('g', Tags.Items.GUNPOWDER)
+			.key('b', WoodenDevices.woodenBarrel)
+			.addCriterion("has_"+toPath(WoodenDevices.woodenBarrel), hasItem(WoodenDevices.woodenBarrel))
+			.build(out);
+
+		ShapedRecipeBuilder.shapedRecipe(WoodenDevices.workbench)
+			.patternLine("ppp")
+			.patternLine("c f")
+			.key('p', IETags.getItemTag(IETags.treatedWood))
+			.key('c', Blocks.CRAFTING_TABLE)
+			.key('f', WoodenDecoration.treatedFence)
+			.addCriterion("has_treated_planks", hasItem(IETags.getItemTag(IETags.treatedWood)))
+			.build(out);
+
+		for(TreatedWoodStyles style:TreatedWoodStyles.values())
+			ShapedRecipeBuilder.shapedRecipe(WoodenDevices.woodenBarrel)
```

Comment:
I would suggest making this a single recipe allowing any mix of styles (by adding a tag for treated slabs).

Reply:
I don't know how to do that yet, i'll play around. See if i can't figure it out just by looking at the already existing stuff. (Same for Below)